pkg/standalone/list.go

```diff
@@ -28,31 +33,78 @@ type ListOutput struct {
 func List() ([]ListOutput, error) {
 	list := []ListOutput{}
 
-	runtimeData, err := rundata.ReadAllRunData()
+	processes, err := ps.Processes()
 	if err != nil {
 		return nil, err
 	}
 
-	for _, runtimeLine := range *runtimeData {
-		proc, err := ps.FindProcess(runtimeLine.PID)
-		if proc == nil && err == nil {
-			continue
-		}
+	for _, proc := range processes {
+		executable := strings.ToLower(proc.Executable())
+		if (executable == "daprd") || (executable == "daprd.exe") {
+			procDetails, err := process.NewProcess(int32(proc.Pid()))
+			if err != nil {
+				continue
+			}
 
-		// TODO: Call to /metadata and validate the runtime data
-		var listRow = ListOutput{
-			AppID:    runtimeLine.AppID,
-			HTTPPort: runtimeLine.DaprHTTPPort,
-			GRPCPort: runtimeLine.DaprGRPCPort,
-			Command:  utils.TruncateString(runtimeLine.Command, 20),
-			Created:  runtimeLine.Created.Format("2006-01-02 15:04.05"),
-			PID:      runtimeLine.PID,
-		}
-		if runtimeLine.AppPort > 0 {
-			listRow.AppPort = runtimeLine.AppPort
+			cmdLine, err := procDetails.Cmdline()
+			if err != nil {
+				continue
+			}
+
+			createUnixTimeMilliseconds, err := procDetails.CreateTime()
+			if err != nil {
+				continue
+			}
+			createTime := time.Unix(createUnixTimeMilliseconds/1000, 0)
+
+			cmdLineItems := strings.Fields(cmdLine)
+			if len(cmdLineItems) <= 1 {
+				continue
+			}
+
+			argumentsMap := make(map[string]string)
+			for i := 1; i < len(cmdLineItems)-1; i += 2 {
+				argumentsMap[cmdLineItems[i]] = cmdLineItems[i+1]
+			}
+
+			httpPort, err := strconv.Atoi(argumentsMap["--dapr-http-port"])
+			if err != nil {
+				continue
+			}
+
+			grpcPort, err := strconv.Atoi(argumentsMap["--dapr-grpc-port"])
+			if err != nil {
+				continue
+			}
+
+			appPort, err := strconv.Atoi(argumentsMap["--app-port"])
+			if err != nil {
+				appPort = 0
+			}
+
+			appID := argumentsMap["--app-id"]
+			appCmd := ""
+			appMetadata, err := metadata.Get(httpPort)
```

Comment:
since we are parsing the cmdline arguments used to launch daprd process. Instead of adding a http get. put on daprd, do you think adding an extra param to daprd like --comment will solve this. The comment can be anything cli wants to put there. This avoids extra api on daprd and extra ipc during the dapr list.

Reply:
> we need more info later one, we would need to keep adding more and more of these "dummy" arguments.

Sure, I was usggesting a --comment param, you decide what to put in it. No future changes needed.

> The flag parsing library in Go only works for command line arguments into Go program, it does not parse an arbitrary string (not big deal, since I assume we can find a solution to this).

daprd is a go program, so it would work.

Are there any other use cases for metadata endpoint? If there are, I am fine with this approach.

Reply:
@amanbha the metadata endpoint is currently giving an actors count for a runtime. I expect the metadata to grow with time.

Reply:
Thats for the get, To enable this, here we are adding a put for metadata, are you saying there could be more scenarios for putting metadata over http

Reply:
> > we need more info later one, we would need to keep adding more and more of these "dummy" arguments.
> 
> Sure, I was usggesting a --comment param, you decide what to put in it. No future changes needed.
> 
> > The flag parsing library in Go only works for command line arguments into Go program, it does not parse an arbitrary string (not big deal, since I assume we can find a solution to this).
> 
> daprd is a go program, so it would work.
> 
Sorry if I was not clear: it works for the command line passed into the program. Here, we are parsing an arbitrary string - is not really the same thing. I've spent some decent amount of time on this.
> Are there any other use cases for metadata endpoint? If there are, I am fine with this approach.

Reply:
> Thats for the get, To enable this, here we are adding a put for metadata, are you saying there could be more scenarios for putting metadata over http

Yes. I can see operator-like things for non k8s-platforms using this in the future.

Reply:
OK, gentlemen, I am convinced, resolving this conversation. Thanks

Reply:
Thanks! This discussion was very clarifying for myself as well.